test/models/compute_resources/compute_resource_test_helpers.rb

```diff
@@ -19,9 +19,108 @@ def mock_cr_servers(cr, servers)
     cr
   end
 
+  def mock_cr(cr, attributes)
+    attributes.each do |attr, stubbed_value|
+      cr.stubs(attr).returns(stubbed_value)
+    end
+    cr
+  end
+
   def assert_find_by_uuid_raises(ex_class, cr)
     assert_raises(ex_class) do
       cr.find_vm_by_uuid('abc')
     end
   end
+
+  def assert_blank_attr_nilified(cr, attr_name)
+    vm_attrs = {
+      attr_name => ''
+    }
+    normalized = cr.normalize_vm_attrs(vm_attrs)
+
+    assert(normalized.has_key?(attr_name))
+    assert_nil(normalized[attr_name])
+  end
+
+  def assert_attrs_mapped(cr, attr_before, attr_after)
+    vm_attrs = {
+      attr_before => 'ATTR_VALUE'
+    }
+    normalized = cr.normalize_vm_attrs(vm_attrs)
+
+    refute(normalized.has_key?(attr_before))
+    assert_equal('ATTR_VALUE', normalized[attr_after])
+  end
+
+  def assert_blank_mapped_attr_nilified(cr, attr_before, attr_after)
+    vm_attrs = {
+      attr_before => ''
+    }
+    normalized = cr.normalize_vm_attrs(vm_attrs)
+
+    refute(normalized.has_key?(attr_before))
+    assert(normalized.has_key?(attr_after))
+    assert_nil(normalized[attr_after])
+  end
+
+  ALLOWED_VM_ATTR_NAMES = %W(
```

Comment:
If plugins use this helper, they will might have a hard time extending this constant.

Reply:
Makes sense, I'll update it.